src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+            tail = head;
+            size++;
+        } else {
+            node = new Node<>(tail, value, null);
+            node.prev.next = node;
+            tail = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexRangeInAdd(index);
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> node = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            if (node.prev == null) {
+                head = newNode;
+            } else {
+                node.prev.next = newNode;
+            }
+            node.prev = newNode;
+            size++;
```

Comment:
You should ensure that the new node is linked from both sides. It seems that the 'next' reference of the new node's previous node is not being set if the new node is inserted at the head of the list (when `node.prev` is null).